src/system/tests/TestSystemPacketBuffer.cpp

```diff
@@ -1873,6 +1873,52 @@ void PacketBufferTest::CheckHandleCloneData(nlTestSuite * inSuite, void * inCont
             config_2.handle = nullptr;
         }
     }
+
+#if CHIP_SYSTEM_PACKETBUFFER_STORE == CHIP_SYSTEM_PACKETBUFFER_STORE_CHIP_HEAP
+
+    // It is possible for a packet buffer allocation to return a larger block than requested (e.g. when using a shared pool)
+    // and in particular to return a larger block than it is possible to request from PackBufferHandle::New().
+    // In that case, (a) it is incorrect to actually use the extra space, and (b) if it is not used, the clone will
+    // be the maximum possible size.
+    //
+    // This is only testable on heap allocation configurations, where pbuf records the allocation size and we can manually
+    // construct an oversize buffer.
+
+    constexpr uint16_t kOversizeDataSize = PacketBuffer::kMaxSizeWithoutReserve + 99;
+    uint8_t buf[PacketBuffer::kStructureSize + kOversizeDataSize];
```

Comment:
Doesn't need to be aligned right to be castable to `PacketBuffer *`?

But better yet, how about we just allocate this with `Platform::MemoryAlloc` which will guarantee alignment and won't need the special `UnsafeRelease` bits either (in that the normal freeing will work correctly)?

Reply:
Done